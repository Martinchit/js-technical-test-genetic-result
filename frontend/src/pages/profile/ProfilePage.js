import React from 'react';
import { Message } from 'semantic-ui-react';
import { InfoItem } from '../../core/components/molecules/InfoItem';
import { FlexBox } from '../../core/components/atoms/FlexBox';
import { Text } from '../../core/components/atoms/Text';
import { LoadingSpinner } from '../../core/components/molecules/LoadingSpinner';
import { useFetchProfile } from '../../core/lib/hooks/useFetchProfile';

export const ProfilePage = () => {
  const [profile, error, loading] = useFetchProfile();

  const headerMapping = React.useMemo(
    () => ({
      id: 'User ID',
      firstName: 'First Name',
      lastName: 'Last Name',
      email: 'Email',
      dateOfBirth: 'Date Of Birth',
      createdAt: 'Created At',
    }),
    []
  );

  console.log(profile, loading, error);

  return (
    <FlexBox $elevation padding="5%" flexDirection="column" backgroundColor="white">
      <Text fontSize={1.8} fontColor="black">
        Profile
      </Text>
      <FlexBox flexDirection="column" margin="10px 0">
        {loading ? (
          <LoadingSpinner inverted />
        ) : !error ? (
          <React.Fragment>
            {Object.entries(profile).map(([k, v]) => (
              <InfoItem
                key={`profile_${k}`}
                header={headerMapping[k]}
                description={v}
              />
            ))}
          </React.Fragment>
        ) : (
          <Message error header="Fetch Error" content="Please refresh" />
        )}
      </FlexBox>
    </FlexBox>
  );
};
