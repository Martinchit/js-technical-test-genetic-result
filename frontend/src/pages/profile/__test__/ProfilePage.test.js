import React from "react";
import { shallow } from 'enzyme';
import * as redux from 'react-redux';

import { ProfilePage } from "../ProfilePage";

describe("ProfilePage", () => {
  let spyOnUseSelector;
  let spyOnUseDispatch;
  let mockDispatch;
  const mockState = {
    auth: {
      token: 'test',
      loggedIn: true
    }
  };
  const waitForAsync = () => new Promise(resolve => setImmediate(resolve))

  beforeEach(() => {
    spyOnUseSelector = jest.spyOn(redux, 'useSelector').mockImplementation(cb => cb(mockState));
    spyOnUseDispatch = jest.spyOn(redux, 'useDispatch');
    mockDispatch = jest.fn();
    spyOnUseDispatch.mockReturnValue(mockDispatch);
  })

  afterEach(() => {
    jest.restoreAllMocks();
  })

  it("renders correctly", () => {
    const wrapper = shallow(<ProfilePage />)
    expect(wrapper).toMatchSnapshot()
  })
})