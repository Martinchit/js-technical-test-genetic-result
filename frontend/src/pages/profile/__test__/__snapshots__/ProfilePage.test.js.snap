// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfilePage renders correctly 1`] = `ShallowWrapper {}`;
